src/init/features/storage.ts

```diff
@@ -3,22 +3,54 @@ import * as clc from "colorette";
 import { logger } from "../../logger";
 import { input } from "../../prompt";
 import { readTemplateSync } from "../../templates";
+import { Config } from "../../config";
+import { Setup } from "..";
+import { FirebaseError } from "../../error";
 
-const RULES_TEMPLATE = readTemplateSync("init/storage/storage.rules");
+export interface RequiredInfo {
+  rulesFilename: string;
+  rules: string;
+  writeRules: boolean;
+}
 
-export async function doSetup(setup: any, config: any): Promise<void> {
-  setup.config.storage = {};
+const RULES_TEMPLATE = readTemplateSync("init/storage/storage.rules");
+const DEFAULT_RULES_FILE = "storage.rules";
 
+export async function askQuestions(setup: Setup, config: Config): Promise<void> {
   logger.info();
   logger.info("Firebase Storage Security Rules allow you to define how and when to allow");
   logger.info("uploads and downloads. You can keep these rules in your project directory");
   logger.info("and publish them with " + clc.bold("firebase deploy") + ".");
   logger.info();
 
-  const storageRulesFile = await input({
+  const info: RequiredInfo = {
+    rulesFilename: DEFAULT_RULES_FILE,
+    rules: RULES_TEMPLATE,
+    writeRules: true,
+  };
+  info.rulesFilename = await input({
```

Comment:
Separate from your PR: IDK why we even ask this in init, we probably should just always use  {current value from firebase.json} || storage.rules

Reply:
Err, we kind of have this pattern of asking rule name.

I guess to confirm it in most cases so people know what this file is for~